src/ripple/rpc/tests/KeyGeneration.test.cpp

```diff
@@ -0,0 +1,190 @@
+
+struct key_strings
+{
+    char const* account_id;
+    char const* master_key;
+    char const* master_seed;
+    char const* master_seed_hex;
+    char const* public_key;
+    char const* public_key_hex;
+    char const* secret_key_hex;
+};
+
+#define PASSPHRASE       "REINDEER FLOTILLA"
+#define MASTER_KEY       "SCAT BERN ISLE FOR ROIL BUS SOAK AQUA FREE FOR DRAM BRIG"
+#define MASTER_SEED      "snMwVWs2hZzfDUF3p2tHZ3EgmyhFs"
+#define MASTER_SEED_HEX  "BE6A670A19B209E112146D0A7ED2AAD7"
+
+#define EXPECT_WALLET_FIELD(r, s, f)  expect (r[#f] == s.f, r[#f].asString())
+
+static key_strings const secp256k1_strings =
+{
+    "r4Vtj2jrfmTVZGfSP3gH9hQPMqFPQFin8f",
+    MASTER_KEY,
+    MASTER_SEED,
+    MASTER_SEED_HEX,
+    "aBQxK2YFNqzmAaXNczYcjqDjfiKkLsJUizsr1UBf44RCF8FHdrmX",
+    "038AAE247B2344B1837FBED8F57389C8C11774510A3F7D784F2A09F0CB6843236C",
+    "1949ECD889EA71324BC7A30C8E81F4E93CB73EE19D59E9082111E78CC3DDABC2",
+};
+
+class WalletPropose_test : public beast::unit_test::suite
+{
+public:
+    void testRandom()
+    {
+        Json::Value params;
+
+        Json::Value result = WalletPropose (params);
+
+        expect (! contains_error (result));
+        expect (result.isMember ("account_id"     ));
+        expect (result.isMember ("master_key"     ));
+        expect (result.isMember ("master_seed"    ));
+        expect (result.isMember ("master_seed_hex"));
+        expect (result.isMember ("public_key"     ));
+        expect (result.isMember ("public_key_hex" ));
+
+        std::string seed = result["master_seed"].asString();
+
+        result = WalletPropose (params);
+
+        expect (result["master_seed"].asString() != seed);
+    }
+
+    void testReindeerFlotilla(Json::Value const& params, key_strings const& s)
+    {
+        Json::Value result = WalletPropose (params);
+
+        expect (! contains_error (result) );
+        EXPECT_WALLET_FIELD (result, s, account_id);
+        EXPECT_WALLET_FIELD (result, s, master_key);
+        EXPECT_WALLET_FIELD (result, s, master_seed);
+        EXPECT_WALLET_FIELD (result, s, master_seed_hex);
+        EXPECT_WALLET_FIELD (result, s, public_key);
+        EXPECT_WALLET_FIELD (result, s, public_key_hex);
+    }
+
+    void testLegacyPassphrase(char const* value)
+    {
+        testcase (value);
+
+        Json::Value params;
+        params["passphrase"] = value;
+
+        testReindeerFlotilla (params, secp256k1_strings);
+    }
+
+    void testLegacyPassphrase()
+    {
+        testLegacyPassphrase (PASSPHRASE);
+        testLegacyPassphrase (secp256k1_strings.master_key);
+        testLegacyPassphrase (secp256k1_strings.master_seed);
+        testLegacyPassphrase (secp256k1_strings.master_seed_hex);
+    }
+
+    void run()
+    {
+        testRandom();
+        testLegacyPassphrase();
+    }
+};
+
+class KeypairForSignature_test : public beast::unit_test::suite
+{
+public:
+    void testEmpty()
+    {
+        Json::Value params;
+
+        bool caught_exception = false;
```

Comment:
Looks like it has `expectException` too: https://github.com/ripple/rippled/blob/develop/src/ripple/basics/TestSuite.h#L75